bluemira/geometry/parameterisations.py

```diff
@@ -1052,108 +1079,90 @@ def _make_domed_leg(
             axis=axis,
-            label="bottom_limb_joint" if flip else "top_limb_joint",
+            label=f"{label}_limb_joint",
         )
 
         # Main leg curve
-        start2 = 90 + np.rad2deg(theta_j)
+        leg_angle = 90 + deg_theta_j
         leg_curve = make_circle(
             radius=r_leg,
             center=leg_centre,
-            start_angle=np.rad2deg(theta_trans) if flip else -start2,
-            end_angle=start2 if flip else -np.rad2deg(theta_trans),
+            start_angle=leg_angle_s(),
+            end_angle=leg_angle_e(),
             axis=[0, 1, 0],
-            label="bottom_limb_dome" if flip else "top_limb_dome",
+            label=f"{label}_limb_dome",
         )
 
         # Outboard corner transition curve
         transition_curve = make_circle(
             radius=r_j,
             center=[x_trans, 0, z_trans],
-            start_angle=0 if flip else -np.rad2deg(theta_trans),
-            end_angle=np.rad2deg(theta_trans) if flip else 0,
+            start_angle=tc_angle_s(),
+            end_angle=tc_angle_e(),
             axis=[0, 1, 0],
-            label="bottom_limb_corner" if flip else "top_limb_corner",
+            label=f"{label}_limb_corner",
+        )
+
+        return BluemiraWire(
+            [corner_in, straight_section, joint_curve, leg_curve, transition_curve][ind],
+            label=f"{label}_limb",
         )
-        # - #
-        # Bluemira wire generation
-        # - #
-        out_wire = []
-        if not flip:
-            out_wire.append(corner_in)
-            out_wire.append(make_polygon([p1, p2]))
-            out_wire.append(joint_curve)
-            out_wire.append(leg_curve)
-            out_wire.append(transition_curve)
-
-        # - #
-        if flip:
-            out_wire.append(transition_curve)
-            out_wire.append(leg_curve)
-            out_wire.append(joint_curve)
-            out_wire.append(make_polygon([p2, p1]))
-            out_wire.append(corner_in)
-        label = "bot_limb" if flip else "top_limb"
-        return BluemiraWire(out_wire, label=label)
 
     @staticmethod
-    def _make_flat_leg(axis, x_in, x_out, z, r_i, r_o, flip=False):
+    def _make_flat_leg(x_in, x_out, z, r_i, r_o, axis=[0, 1, 0], flip=False):
         """
         Makes a flat leg (top/bottom limb) with the option of one end rounded.
 
         Parameters
         ----------
-        axis: np.ndarray
-            [x,y,z] vector normal to plane of parameterisation
-        x_in: np.float
+        x_in: float
             Radial position of inner edge of limb [m]
-        x_out: np.float
+        x_out: float
             Radial position of outer edge of limb [m]
-        z: np.float
+        z: float
             Vertical position of limb [m]
-        r_i: np.float
+        r_i: float
             Radius of inner corner [m]
-        r_o: np.float
+        r_o: float
             Radius of outer corner [m]
+        axis: np.ndarray
```

Comment:
Does this typing need to be as specific as this? Also, the default value is not this type. Perhaps this should be changed to `Iterable[float]` or `Iterable[int]` or similar?